crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs

```diff
@@ -104,14 +106,55 @@ impl Violation for UnusedImport {
 
     fn fix_title(&self) -> Option<String> {
         let UnusedImport { name, multiple, .. } = self;
+        let resolution = match self.context {
+            Some(UnusedImportContext::Init {
+                first_party: true,
+                dunder_all: true,
+            }) => "Add unused import to __all__",
+            Some(UnusedImportContext::Init {
+                first_party: true,
+                dunder_all: false,
+            }) => "Use a redundant alias",
+            _ => "Remove unused import",
+        };
         Some(if *multiple {
-            "Remove unused import".to_string()
+            resolution.to_string()
         } else {
-            format!("Remove unused import: `{name}`")
+            format!("{resolution}: `{name}`")
         })
     }
 }
 
+fn is_first_party(checker: &Checker, qualified_name: &str) -> bool {
```

Comment:
We would typically pass checker as the second argument. We tend to lead with the data that the function is about, and pass extra structs to query as secondary arguments.

Reply:
Willfix